src/Analyzer/Passes/IfTransformStringsToEnumPass.cpp

```diff
@@ -0,0 +1,306 @@
+            return;
+
+        std::string_view function_name = function_node->getFunctionName();
+        if (function_name == "if")
+        {
+            auto & argument_nodes = function_node->getArguments().getNodes();
+
+            if (argument_nodes.size() != 3)
+                return;
+
+            const auto * first_literal = argument_nodes[1]->as<ConstantNode>();
+            const auto * second_literal = argument_nodes[2]->as<ConstantNode>();
+
+            if (!first_literal || !second_literal)
+                return;
+
+            if (!WhichDataType(first_literal->getResultType()).isString() || !WhichDataType(second_literal->getResultType()).isString())
+                return;
+
+            std::set<std::string> string_values;
+            string_values.insert(first_literal->getValue().get<std::string>());
+            string_values.insert(second_literal->getValue().get<std::string>());
+
+            changeIfArguments(argument_nodes[1], argument_nodes[2], string_values, context);
+            return;
+        }
+
+        if (function_name == "transform")
+        {
+            auto & argument_nodes = function_node->getArguments().getNodes();
+
+            if (argument_nodes.size() != 4)
+                return;
+
+            const auto * literal_to = argument_nodes[2]->as<ConstantNode>();
+            const auto * literal_default = argument_nodes[3]->as<ConstantNode>();
+
+            if (!literal_to || !literal_default)
+                return;
+
+            if (!WhichDataType(literal_to->getResultType()).isArray() || !WhichDataType(literal_default->getResultType()).isString())
+                return;
+
+            auto array_to = literal_to->getValue().get<Array>();
+
+            if (array_to.empty())
+                return;
+
+            if (!std::all_of(
+                    array_to.begin(),
+                    array_to.end(),
+                    [](const auto & field) { return field.getType() == Field::Types::Which::String; }))
+                return;
+
+            /// collect possible string values
+            std::set<std::string> string_values;
+
+            for (const auto & value : array_to)
+                string_values.insert(value.get<std::string>());
+
+            string_values.insert(literal_default->getValue().get<std::string>());
+
+            changeTransformArguments(argument_nodes[2], argument_nodes[3], string_values, context);
+            return;
+        }
+    }
+
+private:
+    std::unordered_set<std::string_view> used_functions;
+    ContextPtr context;
+};
+
+}
+
+void IfTransformStringsToEnumPass::run(QueryTreeNodePtr query, ContextPtr context)
+{
+    std::unordered_set<std::string_view> used_functions;
+    std::unordered_set<std::string> function_names{"if", "transform"};
+
+    {
+        FindUsedFunctionsVisitor visitor(used_functions, function_names, 0);
+        visitor.visit(query);
+    }
+
+    {
+        ConvertStringsToEnumVisitor visitor(used_functions, context);
```

Comment:
Consider to move `used_functions` inside `ConvertStringsToEnumVisitor` constructor.